quinn-proto/src/connection.rs

```diff
@@ -651,19 +665,21 @@ impl Connection {
         }
     }
 
-    fn in_recovery(&self, sent_time: u64) -> bool {
+    fn in_recovery(&self, sent_time: Instant) -> bool {
         sent_time <= self.recovery_start_time
     }
 
     fn set_loss_detection_timer(&mut self) {
         if self.in_flight.crypto != 0 || (self.state.is_handshake() && self.side.is_client()) {
             // Handshake retransmission alarm.
-            let timeout = if self.rtt.smoothed == 0 {
+            let timeout = Duration::from_micros(if self.rtt.smoothed == 0 {
                 2 * self.config.initial_rtt
             } else {
                 2 * self.rtt.smoothed
-            };
-            let timeout = cmp::max(timeout, TIMER_GRANULARITY) * 2u64.pow(self.crypto_count);
+            });
+            let timeout = Duration::from_micros(
+                micros_from(cmp::max(timeout, TIMER_GRANULARITY)) * 2u64.pow(self.crypto_count),
+            );
```

Comment:
It would be nicer to use `Duration`'s impl of `Mul<u32>` here; maybe extract `MAX_PTO_EXPONENT` to module scope and reuse it while we're at it.

Reply:
Done.